core/types/transaction.go

```diff
@@ -459,28 +461,56 @@ func (s TxByNonce) Len() int           { return len(s) }
 func (s TxByNonce) Less(i, j int) bool { return s[i].Nonce() < s[j].Nonce() }
 func (s TxByNonce) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 
-type _Order interface {
+type OrderType int
+
+const (
+	Tx OrderType = iota
+	Bundle
+)
+
+type Order interface {
```

Comment:
The order interface can't hold both the non-simulated bundle and the simulated bundles, since it's used for sorting. The way the orders are passed should change, since this interface loses semantic meaning